src/app/icd/ICDCheckInSender.h

```diff
@@ -0,0 +1,58 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <app/icd/ICDMonitoringTable.h>
+#include <credentials/FabricTable.h>
+#include <lib/address_resolve/AddressResolve.h>
+
+namespace chip {
+namespace app {
+
+/**
+ * @brief ICD Manager is responsible of processing the events and triggering the correct action for an ICD
+ */
+class ICDCheckInSender : public AddressResolve::NodeListener
+{
+public:
+    ICDCheckInSender(PersistentStorageDelegate * storage, FabricTable * fabricTable, Crypto::SymmetricKeystore * symmetricKeyStore);
```

Comment:
This needs documentation on what the storage is expected to provide, exactly.

Reply:
actually these can be optimized out